runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -692,9 +692,41 @@ TR_J9ServerVM::getStaticReferenceFieldAtAddress(uintptrj_t fieldAddress)
 bool
 TR_J9ServerVM::stackWalkerMaySkipFrames(TR_OpaqueMethodBlock *method, TR_OpaqueClassBlock *clazz)
    {
+   if(!method)
+      {
+      return false;
+      }
+
    JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
-   stream->write(JITServer::MessageType::VM_stackWalkerMaySkipFrames, method, clazz);
-   return std::get<0>(stream->read<bool>());
+   auto *vmInfo = _compInfoPT->getClientData()->getOrCacheVMInfo(stream);
+   
+   if (vmInfo->_jlrMethodInvoke == NULL)
+      {
+      stream->write(JITServer::MessageType::VM_stackWalkerMaySkipFrames, JITServer::Void());
+      vmInfo->_jlrMethodInvoke = std::get<0>(stream->read<J9Method*>());
+      // stream->write(JITServer::MessageType::VM_stackWalkerMaySkipFrames, method, clazz);
+      // return std::get<0>(stream->read<bool>());
+      }
+   if (vmInfo->_jlrMethodInvoke == ((J9Method *) method))
```

Comment:
After asking the client about the value of `vmInfo->jlrMethodInvoke` you need another check against NULL. According to the original code if `vmInfo->jlrMethodInvoke == NULL`  you can return true